### PR TITLE
Migrationsleitfaden isy-batchrahmen auf isy-security

### PR DESCRIPTION
**Pull request by FabianPerl**
Depends on #46

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -50,6 +50,9 @@
 - `IFS-2153`: [isyfact-task] Umstellung isy-task auf Spring Boot:
   #### _BREAKING CHANGE: Einführung von Spring Boot_
   * Taskkonfiguration und TaskkonfigurationVerwalter entfernt, Übersetzungen hinzugefügt
+- `IFS-1853`: [isy-batchrahmen] Umstellung von `isy-sicherheit` und `isy-aufrufkontext` auf neuen Baustein `isy-security`
+  #### _Breaking Change:_
+  * Die Konfiguration von `benutzer`, `passwort` und `bhknz` erfolgt über isy-security ClientRegistrations. Batches müssen nur noch eine `registrationId` zur Authentifizierung bereitstellen. Die Methode `getAuthenticationCredentials` der `BatchAusfuehrungsBean` wurde entfernt.
```
Comment (kollmana) [inline CHANGELOG.md]: Was registrationId not renamed to'oauth2ClientRegistrationId'?

File: isy-batchrahmen/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,9 @@
 - `IFS-1169`: Änderung von Tabellenname und Spaltennamen der BatchStatus-Tabelle zu `CamelCaseToUnderscoresNamingStrategy`
     * Tabelle `BATCH_STATUS` mit `CamelCaseToUnderscoresNamingStrategy` ersetzt Tabelle `BATCHSTATUS` mit `PhysicalNamingStrategyStandardImpl`
     * explizites ORM für Abwärtskompatibilität bereitgestellt (siehe Dokumentation: Die Konfiguration der Spring-Kontexte)
+- `IFS-1853`: Umstellung von `isy-sicherheit` und `isy-aufrufkontext` auf neuen Baustein `isy-security`
+  #### _Breaking Change:_
+  * Die Konfiguration von `benutzer`, `passwort` und `bhknz` erfolgt über isy-security ClientRegistrations. Batches müssen nur noch eine `registrationId` zur Authentifizierung bereitstellen. Die Methode `getAuthenticationCredentials` der `BatchAusfuehrungsBean` wurde entfernt.
```
Comment (kollmana) [inline isy-batchrahmen/CHANGELOG.md]: Was registrationId not renamed to'oauth2ClientRegistrationId'?

File: isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/detailkonzept-komponente-batch/inhalt.adoc [hunk 3]
```diff
@@ -738,19 +738,9 @@ Datei `JRE_HOME/lib/management/jmxremote.access` readwrite-Zugriff erlaubt ist.
 Ein Batch wird innerhalb der Plattform gestartet.
 Dazu verwendet der Batch die Benutzerkennung eines technischen Benutzers „internes System“, die in seiner Startkonfiguration (als Konfigurationsdatei) hinterlegt und im Benutzerverzeichnis vorhanden ist.
 
-Der Batchrahmen nutzt die T-Komponente Sicherheit (siehe xref:isy-sicherheit:nutzungsvorgaben/master.adoc#einleitung[Nutzungsvorgaben Sicherheit]), um den anfragenden Benutzer über den IAM-Service der Plattform zu authentifizieren.
-Das IAM-Service authentifiziert den Benutzer anhand der angegebenen Benutzerkennung und des Passwortes, registriert alle Informationen zum Benutzer in der T-Komponente AufrufKontextVerwalter und schließt die Session des Benutzers im IAM-System.
+Der Batchrahmen verwendet den Baustein `isy-security`, um den anfragenden Benutzer über einen der Authentifizierungswege Client Credential oder Ressource Owner Password (deprecated) via `registrationId` zu authentifizieren.
```
Comment (kollmana) [inline isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/detailkonzept-komponente-batch/inhalt.adoc]: Was registrationId not renamed to'oauth2ClientRegistrationId'?

File: isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/detailkonzept-komponente-batch/inhalt.adoc [hunk 4]
```diff
@@ -759,19 +749,15 @@ Folgende Informationen zum Benutzer sind anzugeben
 [cols="2,8",options="header"]
 |====
 |Property |Beschreibung
-|`Bhknz` |Das Behörden- oder Organisationskennzeichen, der Behörde oder Organisation, der der Benutzer angehört.
-|`Benutzer` |Die Kennung des technischen Benutzers (internes System)
-|`Passwort` |Das Kennwort des Benutzers
+|`registrationId` |Die ID der Client Registration aus `isy-security`, die zur Authentifizierung genutzt wird
```
Comment (kollmana) [inline isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/detailkonzept-komponente-batch/inhalt.adoc]: Was registrationId not renamed to'oauth2ClientRegistrationId'?

File: isyfact-standards-doc/src/docs/antora/modules/changelog/pages/migrationsleitfaden.adoc [hunk 5]
```diff
@@ -17,6 +17,65 @@ Die auf dieser Seite aufgeführten Hinweise sollen IsyFact Anwendungsentwicklern
...
+batch.bhknz=TestBHKNZ
+
+# [...]
+----
+
+[[batch-configuration-isy-security]]
+.Batch-Konfiguration zur Verwendung von isy-security in IsyFact 3
+[source,properties]
+----
+batch.registrationId=TestClient
```
Comment (kollmana) [inline isyfact-standards-doc/src/docs/antora/modules/changelog/pages/migrationsleitfaden.adoc]: Was registrationId not renamed to'oauth2ClientRegistrationId'?

File: isyfact-standards-doc/src/docs/antora/modules/changelog/pages/migrationsleitfaden.adoc [hunk 6]
```diff
@@ -17,6 +17,65 @@ Die auf dieser Seite aufgeführten Hinweise sollen IsyFact Anwendungsentwicklern
...
+
+# [...]
+
+spring.security.oauth2.client.provider.TestRealm.issuer-uri=http://localhost:9095/auth/realms/testrealm
+spring.security.oauth2.client.registration.TestClient.client-id=test-user-client-id
+spring.security.oauth2.client.registration.TestClient.client-secret=test-user-client-secret
+spring.security.oauth2.client.registration.TestClient.authorization-grant-type=client_credentials
+spring.security.oauth2.client.registration.TestClient.provider=TestRealm
+----
```
Comment (kollmana) [inline isyfact-standards-doc/src/docs/antora/modules/changelog/pages/migrationsleitfaden.adoc:78]: I am missing an explanation of the (new) configuration parameters: 
- What is the issuer-uri?
- Why does a technical user has to have a client-id & -secret and where do they come from?

Comment (kollmana) [inline isyfact-standards-doc/src/docs/antora/modules/changelog/pages/migrationsleitfaden.adoc:78]: What are the steps for an application to retrieve their actual credentials?

**Review by kollmana — COMMENTED** [review]
Migrationsleitfaden isy-batchrahmen: 
- -Was registrationId not renamed to'oauth2ClientRegistrationId'?-
- -Explanation of new security config parameters-

**Review by kollmana — COMMENTED** [review]
-Was registrationId not renamed to'oauth2ClientRegistrationId'?-

I am missing an explanation of the (new) configuration parameters:
- What is the issuer-uri?
- Why does a technical user has to have a client-id & -secret and where do they come from?

File: CHANGELOG.md (hunk 1 above)
Comment (FabianPerl) [reply CHANGELOG.md]: This is not part of this pull request. Nevertheless, the branch here has been adapted to the new state of IFS-1853

File: isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/detailkonzept-komponente-batch/inhalt.adoc (hunk 4 above)
Comment (FabianPerl) [reply isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/detailkonzept-komponente-batch/inhalt.adoc]: This is not part of this pull request. Nevertheless, the branch here has been adapted to the new state of IFS-1853

File: isyfact-standards-doc/src/docs/antora/modules/changelog/pages/migrationsleitfaden.adoc (hunk 5 above)
Comment (FabianPerl) [reply isyfact-standards-doc/src/docs/antora/modules/changelog/pages/migrationsleitfaden.adoc]: Adapted to new variable name

File: isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/detailkonzept-komponente-batch/inhalt.adoc (hunk 3 above)
Comment (FabianPerl) [reply isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/detailkonzept-komponente-batch/inhalt.adoc]: Adapted to new variable name

File: isy-batchrahmen/CHANGELOG.md (hunk 2 above)
Comment (FabianPerl) [reply isy-batchrahmen/CHANGELOG.md]: This is not part of this pull request. Nevertheless, the branch here has been adapted to the new state of IFS-1853

**Comment by FabianPerl** [comment]
For detailed instructions on how to migrate I refer you to [PR 66](https://github.com/IsyFact/isyfact-standards/pull/66), here in this ticket it is only described how to use isy-security in isy-batchrahmen.

**Review by kollmana — APPROVED** [review]

**Pull request merged**